### ISSUE TITLE
Revisar Figma como preparação para implementação da busca

### ISSUE THREAD

**Issue by anapaulagomes** [DadosAbertosDeFeira/portal#139]
Figma: https://www.figma.com/file/XNfcVKFQLMcw4WzYB9srDs/DadosFeira?type=design&node-id=224-2144&t=Rhddys1UH4rsOZPn-0

Aqui nessa issue vamos documentar quais as dúvidas e o que precisamos para implementação da busca por parte do design.

**Comment by matheusrocha89** [comment]
As dúvidas que hoje eu tenho são: 

1 - A API suporta buscas por essas categorias? 
<img width="368" alt="image" src="https://github.com/DadosAbertosDeFeira/portal/assets/3718366/6e178e08-c41b-48b1-8b57-7b2ba3abf8a4">
Se sim, existe um endpoint onde podemos buscar essas categorias ou vai ser hard coded?

2 - Nossa API já tem paginação para os resultados?

3 - A pré-vizualização é um PDF? 
<img width="881" alt="image" src="https://github.com/DadosAbertosDeFeira/portal/assets/3718366/1d0a07c0-117a-4620-b28c-84ea603872c6">

4 - Falta uma página para quando não tiver resultados e uma para caso ocorra algum erro, eu não vi design dessas?

Por enquanto tenho só essas dúvidas 😄

**Comment by anapaulagomes** [comment]
A respeito da API:
* devo trabalhar nessa issue assim que voltar de férias pra que vocês possam testar e ver os endpoints que temos ao vivo. Por enquanto temos essa documentação aqui (que está desatualizada).
* a API tem paginação sim

Sobre o design:
* acho que podemos tirar o PDF como visualização; as pessoas podem baixar se acharem q precisam
* bom ponto sobre a página pra caso a busca não retorne nada 👍🏽

**Comment by anapaulagomes** [comment]
Uma coisa que não vi:
* um lugar para quantos resultados foram encontrados

Dúvida:
* vocês acham que faz sentido adicionar logo de cara a opção de uma busca global (selecionando várias bases de dados ou todos os arquivos que temos)? tipo, uma pessoa quer pesquisar um contrato pelo código. Essa informação pode estar no diário, nas licitações e nos contratos. Ou até mesmo nomes, de pessoas ou empresas... O que acham?

**Comment by anapaulagomes** [comment]
@matheusrocha89 acho que a busca é essa aqui

![Screenshot 2023-05-24 at 19 55 41](https://github.com/DadosAbertosDeFeira/portal/assets/1899950/7d48ad82-a733-4908-920d-d94b878e2cec)

Está meio confuso de saber mesmo. Acho que precisamos organizar melhor esse Figma. 😅

**Comment by matheusrocha89** [comment]
> @matheusrocha89 acho que a busca é essa aqui
> 
> ![Screenshot 2023-05-24 at 19 55 41](https://user-images.githubusercontent.com/1899950/240701133-7d48ad82-a733-4908-920d-d94b878e2cec.png)
> 
> Está meio confuso de saber mesmo. Acho que precisamos organizar melhor esse Figma. 😅

Mas essas opções a gente puxa do BE certo? Não vamos colocar hard coded porque se o valor dessas opções mudar ou a gente precisar criar mais ou tirar alguma fica consistente essa lógica em um só lugar.

**Comment by anapaulagomes** [comment]
Faz sentido, @matheusrocha89! Vou criar uma issue pra gente colocar isso num endpoint do backend, já com um nome amigável.

A busca global me preocupa em termos de performance mas não sei se teremos tantos acessos simultâneos assim (depende sempre do momento e do tema). Mas vamos testando! Se não der colocamos uma flag pra incluir busca em arquivos. O resto é tranquilo.

**Comment by matheusrocha89** [comment]
Concordo, vamos adaptando com a necessidade. No futuro se precisarmos de performance temos várias soluções que podemos usar e isso é um problema do futuro, não de agora 😄 Se der merda, na dúvida colocamos uma feature flag que desliga a opção global
